src/TestData/Templates/Package.wxs

```diff
@@ -21,6 +21,9 @@
     <Feature Id="Complete" Level="1">
       <ComponentRef Id="FileComponent" />
       <ComponentRef Id="RegistryComponent" />
+      <?ifdef var.ProductComponents?>
```

Comment:
It would explain the last two errors if it's set for PackageAv1.wixproj but not PackageAv1_0_1.wixproj. I'd start by diffing the two MSIs debugging the unit test. If they're the same, it's probably @robmen's fault in the patching code.

Reply:
That's set for both in PackageA.props which both .wixprojs import.

https://github.com/rseanhall/burn-e2e-tests/blob/2d3ea8d39152bd7590d067c64fb64d9999a8a8de/src/TestData/PatchTests/PackageAv1/PackageA.props#L5

Reply:
Both registry components have different ids in v1 vs v1.0.1.

Reply:
Yep, I said it *would* explain it, omitting that it would then create a new mystery. 🥇

Reply:
Yeah, no idea. That's wacky given that all the preprocessor variables are the same.

Reply:
So the registry ComponentIds are supposed to be stable, right? That's the bug here? Because RegistryComponent has GUID1 in v1 and GUID2 in v1.0.1. And RegistryComponent2 has GUID3 in v1 and GUID4 in v1.0.1.

Reply:
Problem is that the guid for a registry-keypath component uses Key, Name, and Value. Value changes between the two builds. I'm wondering if it's supposed to be Root, Key, and Name instead.

Reply:
Fields 0..2 -v- 1..3.